packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -73,9 +73,23 @@ function DataClassEditor(props: EditorProps) {
   const dataClass = data.data;
   const dataClassFields = dataClass.fields;
 
-  const title = isEntityClass(dataClass) ? 'Entity Class Editor' : 'Data Class Editor';
-  let detailTitle = title;
-  if (selectedField !== undefined && selectedField < dataClassFields.length) {
+  let baseTitle = '';
+  switch (classType(dataClass)) {
+    case 'DATA':
+      baseTitle = 'Data';
+      break;
+    case 'BUSINESS_DATA':
+      baseTitle = 'Business Data';
+      break;
+    case 'ENTITY':
+      baseTitle = 'Entity';
+  }
+  const title = `${baseTitle} Editor`;
+
+  let detailTitle = '';
+  if (selectedField === undefined) {
+    detailTitle = `${baseTitle} - ${dataClass.simpleName}`;
+  } else if (selectedField < dataClassFields.length) {
```

Comment:
If you move this code to it's own function, you can easily test it with a unit test which should be definilty enough